test/unit/gateways/global_collect_test.rb

```diff
@@ -36,23 +36,22 @@ def test_successful_authorize_and_capture
   end
 
   def test_successful_purchase_with_requires_approval_true
+    call_count = 0
     stub_comms do
       @gateway.purchase(@accepted_amount, @credit_card, @options.merge(requires_approval: true))
     end.check_request do |endpoint, data, headers|
-      assert_equal true, JSON.parse(data)['cardPaymentMethodSpecificInput']['requiresApproval']
-    end.respond_with(successful_authorize_response)
+      call_count += 1
+      assert_equal(true, JSON.parse(data)['cardPaymentMethodSpecificInput']['requiresApproval']) if call_count == 1
+    end.respond_with(successful_authorize_response, successful_capture_response)
+    assert call_count == 2 # auth and capture
```

Comment:
Is there a less janky way to do this line and line 44? 🤔

Reply:
Just a suggestion, but I think the fact that it's returning both the `successful_authorize_response` and the `successful_capture_response` is sufficient proof that it's behaving the way it should even without explicitly checking for the `requiresApproval` field. It would certainly be less janky although a tad less thorough.